### DISCUSSION TITLE
Maintain multiple Inertia apps in the same Laravel installation

### DISCUSSION THREAD

**Discussion by craigrileyuk** [inertiajs/inertia#1051] [Help]
Is there any way to maintain two completely separate Inertia applications within the same Laravel installtion?

Say, for example, you want to have a backend application in Inertia, then also a frontend in Inertia too.

Obviously, you can get both to work out the box, but then you'd have the same vendor files, same default blade file, the same Inertia::share data and potentially a lot of css and js bloat from sections of the site which would be completely unconnected.

Ideally, I'd like to maintain separate folders and package.json files for both Inertia instances.

Possible?

**Comment by rhysemmerson** [comment]
I think it would be possible if you create 2 Inertia middlewares, one for the frontend and one for the backend, you'd add the middlewares as named middlewares rather than default ones and add them to your frontend/backend route groups respectively. 

You could set the rootView property of each to be a different view that includes different js and css files. You'd then configure mix to build those separate bundles.
